examples/automm/cache_adapter/cache_adapter.py

```diff
@@ -0,0 +1,465 @@
+    parser.add_argument("--train_epoch_F", type=int, default=20, help="The training epochs for finetuning the cache adapter.")
+    parser.add_argument("--init_alpha", type=float, default=1.17, help="The initial value of hyper-parameter alpha in cache adapter")
+    parser.add_argument("--init_beta", type=float, default=1., help="The initial values of hyper-parameter beta in cache adapter.")
+    parser.add_argument("--search_scale", type=list, default=[10, 10], help="The search scale of alpha and beta.")
+    parser.add_argument("--search_step", type=list, default=[200, 20], help="The steps of searching hyper-parameters alpha and beta.")
+    args = parser.parse_args()
+
+    args.cache_dir = os.path.join('./caches', args.dataset)
+    os.makedirs(args.cache_dir, exist_ok=True)
+
+    if args.column_names is None:
+        args.column_names = ["text"] if args.type == "text" else ["image"]
+    
+    if args.backbone is None:
+        if args.type == "text":
+            args.backbone = "sentence-transformers/paraphrase-mpnet-base-v2"
+        elif args.type == "image":
+            args.backbone = "swin_base_patch4_window7_224"
+        else:
+            args.backbone = "openai/clip-vit-large-patch14-336"
+    
+    return args
+
+
+class AutoMMCacheAdapter(nn.Module):
+    """
+    The model to generate few shot predict proba with 
+    features extracted by AutoGluon MultiModal Predictor.
+    """
+
+    def __init__(
+        self, 
+        cache_keys, 
+        cache_values,
+        hidden_size, 
+        num_classes,
+        clip_weights=None, 
+    ):
+        """
+        Create the model head and the cache adapter.
+
+        Parameters
+        ----------
+        cache_keys
+            The content of cache composed of features in the training set.
+        cache_values 
+            The one-hot encoded label of corresponding cache_keys.
+        hidden_size
+            The size of features.
+        num_classes
+            The classes of the dataset.
+        clip_weights
+            The encoded label-text prompts which is only used in type "clip".
+        """
+        super(AutoMMCacheAdapter, self).__init__()
+        self.cache_keys = cache_keys
+        self.cache_values = cache_values
+        self.adapter = nn.Linear(cache_keys.shape[0], cache_keys.shape[1], bias=False)
+        self.adapter.weight = nn.Parameter(cache_keys.t())
+
+        self.clip_weights = clip_weights
+
+        self.model_head = nn.Linear(hidden_size, num_classes, bias=True) if clip_weights is None else None
+
+    def get_adapter_logits(self, affinity, pure_logits, alpha, beta):
+        """
+        Generate logits with cache adapter based on pure_logits and cache output.
+
+        Parameters
+        ----------
+        affinity
+            The result of cache similarity. It is based on cosine similarity or a projector initialized with cache_keys.
+        pure_logits
+            The predict proba of the classifier.
+        alpha
```

Comment:
Both alpha and beta are shown as hyperparameters of cache model. What do they represent? Do they have default values? Please add more details to the documentation, thank you.